x-pack/elastic-agent/pkg/artifact/install/zip/zip_installer.go

```diff
@@ -120,6 +120,11 @@ func (i *Installer) unzip(artifactPath string) error {
 	}
 
 	for _, f := range r.File {
+		// if we were cancelled in between
+		if err := ctx.Err(); err != nil {
+			return err
```

Comment:
Won't this still leave it broken? If the context is cancelled, then this will stop extracting.

Reply:
this will propagate error and prevent install to continue

Reply:
This will prevent all of the files in the zip file from being extracted. So that mean that if a SIGTERM occurs it will cancel the context. When the context is cancelled and its in this loop then only part of the zip file will be extracted.

What happens when this returns a `context.Cancelled`? Does the zip installer then remove the directory because the context was cancelled during the extraction? If that is the case, then this is okay. If it accepts `context.Cancelled` as an acceptable error, or does not remove the directory then it would still be left in a bad state.

Reply:
yes that's the case, atomic installer removes the extracted files because error was returned